### PR TITLE
Add renv lockfiles into object and use renv::install for external deps if inside renv environment

### LINKED ISSUE #146: Use `renv::install` when working in an renv environment
If you are working inside an renv environment when you call `install_deps` then any missing external dependencies are installed using `remotes` (specifically `remotes::install_deps`) and it appears `remotes`  does not use the renv cache:

image

Using `renv::install` installs dependencies and does use the cache:

image

image

We should allow (enable by default?) using the renv cache if you are in an active renv project...

### PR DESCRIPTION
**Pull request by nikolas-burkoff**
Closes: https://github.com/openpharma/staged.dependencies/issues/147 and closes #146 
image

To test #147:

```
x <- dependency_table("openpharma/stageddeps.house", "repo@host", ref = "renv@main")
print(x$renv_files$stageddeps.house)
print(x$renv_files$stageddeps.food)
# test a profile which only exists on stageddeps.house:
x <- dependency_table("openpharma/stageddeps.house", "repo@host", ref = "profiles@renv@main", renv_profile = "example")
print(x$renv_files$stageddeps.house)
print(x$renv_files$stageddeps.food)
```

To test #146 

First run - outside of an renv environment:

```
x <- dependency_table("openpharma/stageddeps.house", "repo@host", ref = "renv@main") # could use "main" 
install_deps(x)
```

Then create an renv environment and make sure Bioconductor repos are in options("repos") and run the same command and you'll see the dependencies coming from the cache (or being installed into the cache if they are not already in the cache)

### CODE REVIEW

File: R/rstudio_jobs.R [hunk 1]
```diff
@@ -24,7 +24,8 @@ run_job <- function(text, tempfile_prefix = "file", jobname_prefix = "Job", ...)
 #' install_deps_job(create_args = list(direction = "all"))
 #' install_deps_job(dry_install = TRUE)
 #' }
-install_deps_job <- function(project = ".", project_type = "local", verbose = 1, create_args = list(), ...) {
+install_deps_job <- function(project = ".", project_type = "local", verbose = 1,
+                             create_args = list(renv_profile = Sys.getenv("RENV_PROFILE")), ...) {
```
Comment (nikolas-burkoff) [inline R/rstudio_jobs.R:28]: When running the jobs (and therefore the addins) we take the current active renv_profile (if there is one) - this is in contrast to when running the `dependency_table` function where the default is to use renv_profile = NULL - see dependency_table function

**Comment by pawelru** [comment]
I can see that there are a lot of dynamic creation of files/dirs paths. Have you seen `renv::paths` already? You might find there something you are trying to create on your own.

**Comment by nikolas-burkoff** [comment]
> can see that there are a lot of dynamic creation of files/dirs paths. Have you seen renv::paths already? You might find there something you are trying to create on your own.

So if you run `dependency_table("openpharma/x", "repo@host", ref = "main")` then a cloned copy of each repo, including "openpharma/x" is checked out into the cache which is fine - but if you run the addin or start from a local copy of a repo (i.e `dependency_table("../my_files/x")` then staged.deps copies your local copy of the repo into the cache.

When we copy we don't want to copy the whole renv library as that is huge and takes ages but we do want to copy over all the renv.lock files (some of which are in renv/profiles/something/ directory) (these are there if you did a fresh clone) - hence the messing around with paths to make sure the right things are copied over

File: R/git_tools.R [hunk 2]
```diff
@@ -174,10 +175,18 @@ install_external_deps <- function(repo_dir, internal_pkg_deps, ...) {
...
+
+  #also change name to make it clearer for users when using renv::install
+  desc_obj$set(Package = "staged.deps.tmp.pkg")
   desc_obj$set_deps(new_deps)
   desc_obj$write()
 
-  remotes::install_deps(repo_dir_external, ...)
+  if (!is.null(Sys.getenv("RENV_PROJECT")) && Sys.getenv("RENV_PROJECT") != "" && requireNamespace("renv", quietly = TRUE)) {
+    renv::install(repo_dir_external)
+    renv::remove("staged.deps.tmp.pkg")
```
Comment (nikolas-burkoff) [inline R/git_tools.R:186]: May be able to remove line 180 and 186 and it still work and be less confusing for users?

Comment (nikolas-burkoff) [reply R/git_tools.R]: yup these can and have been removed

File: NEWS.md [hunk 3]
```diff
@@ -1,5 +1,8 @@
 # staged.dependencies 0.2.5 (unreleased)
```
Comment (nikolas-burkoff) [inline NEWS.md:1]:
```suggestion
# staged.dependencies 0.2.5 
```

File: R/dependencies.R [hunk 4]
```diff
@@ -23,6 +23,8 @@
 #'   either "upstream","downstream" or "all".
 #' @param fallback_branch (`character`) the default branch to try to use if
 #'   no other matches found
+#' @param renv_profile (`character`) the name of the renv profile of the renv.lock files
```
Comment (gogonzo) [inline R/dependencies.R:26]:
```suggestion
#' @param renv_profile (`character`) the name of the renv profile of the `renv.lock` files
```

File: R/dependencies.R [hunk 5]
```diff
@@ -23,6 +23,8 @@
...
+#'   to be included from the repos. The standard renv.lock file uses the default NULL argument here.
```
Comment (gogonzo) [inline R/dependencies.R:27]:
```suggestion
#'   to be included from the repos. The standard `renv.lock` file uses the default `NULL` argument here.
```

File: R/dependencies.R [hunk 6]
```diff
@@ -44,6 +46,8 @@
 #'               R packages found in the description files of the internal packages. It is a dataframe
 #'               of the form returned by `desc::desc_get_deps`}
 #'   \item{direction}{`direction` argument used to create object}
+#'   \item{renv_files}{`named list` containing the json of the renv.lock files for the chosen profile for
+#'                     each repo. An entry to the list is NULL if a repos does not have the required lock file}
```
Comment (gogonzo) [inline R/dependencies.R:50]:
```suggestion
#'                     each repo. An entry to the list is `NULL` if a repos does not have the required lock file}
```

File: R/dependencies.R [hunk 7]
```diff
@@ -317,14 +326,17 @@ plot.dependency_structure <- function(x, y, ...){
 #' @param install_direction "upstream", "downstream" or "all"; which packages
 #'   to install (according to dependency structure). By default this is only "upstream"
 #' @param install_external_deps logical to describe whether to install
-#'   external dependencies of packages using `remotes::install_deps`.
-#' @param upgrade argument passed to `remotes::install_deps`, defaults to 'never'.
+#'   external dependencies of package using `remotes::install_deps` (or `renv::install` if
```
Comment (gogonzo) [inline R/dependencies.R:329]:
```suggestion
#'   external dependencies of package using [remotes::install_deps()] (or [renv::install()] if
```

File: R/dependencies.R [hunk 8]
```diff
@@ -317,14 +326,17 @@ plot.dependency_structure <- function(x, y, ...){
...
+#'   inside an renv environment) .
+#' @param upgrade argument passed to `remotes::install_deps`, defaults to 'never'. Ignored
```
Comment (gogonzo) [inline R/dependencies.R:331]:
```suggestion
#' @param upgrade argument passed to [remotes::install_deps()], defaults to `'never'`. Ignored
```

File: R/dependencies.R [hunk 9]
```diff
@@ -317,14 +326,17 @@ plot.dependency_structure <- function(x, y, ...){
...
+#'   if inside an `renv` environment.
 #' @param package_list (`character`) If not NULL, an additional filter, only packages on this
 #'   list will be considered and their dependencies installed if needed (advanced usage only).
 #' @param dry (`logical`) dry run that outputs what would happen without actually
 #'   doing it.
 #' @param verbose verbosity level, incremental; from 0 (none) to 2 (high)
-#' @param ... Additional args passed to `remotes::install_deps.
+#' @param ... Additional args passed to `remotes::install_deps. Ignored
```
Comment (gogonzo) [inline R/dependencies.R:338]:
```suggestion
#' @param ... Additional args passed to [remotes::install_deps()]. Ignored
```

File: R/dependencies_app.R [hunk 10]
```diff
@@ -14,6 +14,8 @@
 #' @param default_ref (`character`) default ref (branch/tag), see also the parameter
 #'   `ref` of `\link{dependency_table}`. If `NULL` this must be entered by app user
 #'   and can always be changed by the user.
+#' @param renv_profile (`character`) the name of the renv profile of the renv.lock files
```
Comment (gogonzo) [inline R/dependencies_app.R:17]:
```suggestion
#' @param renv_profile (`character`) the name of the `renv` profile of the `renv.lock` files
```

File: R/dependencies_app.R [hunk 11]
```diff
@@ -14,6 +14,8 @@
...
+#'   to be included from the repos. The standard renv.lock file uses the default NULL argument here.
```
Comment (gogonzo) [inline R/dependencies_app.R:18]:
```suggestion
#'   to be included from the repos. The standard `renv.lock` file uses the default `NULL` argument here.
```

File: R/dependencies_helper.R [hunk 12]
```diff
@@ -13,15 +13,18 @@
 #' @param internal_pkg_deps packages to not install (when `install_external_deps = TRUE`)
 #' @param dry logical, if `FALSE` (default) run the actions, if `TRUE` do not
 #' @param install_external_deps logical to describe whether to install
-#'   external dependencies of package using `remotes::install_deps`.
-#' @param upgrade argument passed to `remotes::install_deps`, defaults to 'never'.
+#'   external dependencies of package using `remotes::install_deps` (or `renv::install` if
+#'   inside an renv environment) .
+#' @param upgrade argument passed to `remotes::install_deps`, defaults to 'never'. Ignored
+#'   if inside an `renv` environment.
```
Comment (gogonzo) [inline R/dependencies_helper.R:19]:
```suggestion
#'   external dependencies of package using [remotes::install_deps()] (or [renv::install()] if
#'   inside an `renv` environment) .
#' @param upgrade argument passed to [remotes::install_deps()], defaults to `'never'`. Ignored
#'   if inside an `renv` environment.
```

File: R/dependencies_helper.R [hunk 13]
```diff
@@ -13,15 +13,18 @@
...
 #' @param rcmd_args list with names `build`, `check`,
 #'   `install` which are vectors that are passed as separate arguments
 #'   to the `R CMD` commands
 #' @param artifact_dir directory to store log files, only when actions include
 #'   `build`; action `test` only outputs to the console
 #' @param verbose verbosity level, incremental - from 0 (none) to 2 (high)
-#' @param ... Additional args passed to `remotes::install_deps.
+#' @param ... Additional args passed to `remotes::install_deps Ignored
```
Comment (gogonzo) [inline R/dependencies_helper.R:26]:
```suggestion
#' @param ... Additional args passed to [remotes::install_deps()] Ignored
```

**Review by gogonzo — COMMENTED** [review]
Couple of documentation fixes suggested. More important is to upgrade DESCRIPTION to include `Roxygen: list(markdown = TRUE)`

**Review by gogonzo — APPROVED** [review]

**Pull request merged**